### PR TITLE
feature: disconnect from dapp

### PR DESCRIPTION
**Pull request by jasonzhouu**
Implements  https://github.com/argentlabs/argent-x/issues/262

https://user-images.githubusercontent.com/19926174/155849631-67924e4e-4618-4153-a907-f38f0981a87d.mov

https://user-images.githubusercontent.com/19926174/155849796-7e4bf8e1-819d-4914-a357-95b0a5d66652.mov

## PS
npm package `@argent/get-starknet` should be upgraded in the Dapp.

### CODE REVIEW

File: packages/extension/src/background/index.ts [hunk 1]
```diff
@@ -189,6 +190,21 @@ import { addToWhitelist, isOnWhitelist } from "./whitelist"
         return openUi()
       }
 
+      case "DISCONNECT": {
+        const { host } = msg.data
+        console.log("sender", sender)
+        console.log("host", host)
```
Comment (delaaxe) [inline packages/extension/src/background/index.ts:196]: console.logs

File: packages/extension/src/shared/MessageType.ts [hunk 2]
```diff
@@ -23,7 +23,9 @@ export type MessageType =
   | { type: "GET_SELECTED_WALLET" }
   | { type: "GET_SELECTED_WALLET_RES"; data: BackupWallet }
   | { type: "CONNECT"; data: { host: string } }
+  | { type: "DISCONNECT"; data: { host: string } }
   | { type: "CONNECT_RES"; data: BackupWallet }
+  | { type: "DISCONNECT_RES" }
```
Comment (delaaxe) [inline packages/extension/src/shared/MessageType.ts:28]: The order should stay:
EVENT
EVENT_RES
EVENT2
EVENT2_RES

File: packages/get-starknet/src/index.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ const defaultStarknetWindowObject: StarknetWindowObject = {
     }
     return fail()
   },
+  disconnect: async () => {},
```
Comment (delaaxe) [inline packages/get-starknet/src/index.ts:23]: Should be like others:

```suggestion
  disconnect: fail,
```

**Review by delaaxe — CHANGES_REQUESTED** [review]

**Comment by jasonzhouu** [comment]
> See the comment I made on the issue

You mean implement 'chainChanged', 'connect', 'disconnect'?

**Comment by jasonzhouu** [comment]
It seems that Metamask don't emit 'disconnect' event when use disconnect dapp from Metamask.

**Comment by jasonzhouu** [comment]
Metamask's 'disconnect' event is emitted when RPC is disconnected:

> In general, this will only happen due to network connectivity issues or some unforeseen error.

https://docs.metamask.io/guide/ethereum-provider.html#disconnect

**Comment by jasonzhouu** [comment]
It seems that the request type of the develop branch is a bit different, I'll check it out tomorrow

**Comment by jasonzhouu** [comment]
> i was referencing this comment: [#262 (comment)](https://github.com/argentlabs/argent-x/issues/262#issuecomment-1054294025)

This PR does not act like JediSwap. It seems that action of disconnection in JediSwap does not interact with ArgentX.

**Comment by delaaxe** [comment]
Yes. Disconnecting should not remove from whitelist

**Comment by jasonzhouu** [comment]
Does argentx needs to record when user disconnect dapp from argentx, like what Metamask did?

**Comment by gergold** [comment]
This should be covered by #378 
So closing this PR

**Pull request closed without merging**